usecases/traverser/hybrid/searcher.go

```diff
@@ -176,6 +176,49 @@ func Search(ctx context.Context, params *Params, logger logrus.FieldLogger, spar
 	return fused, nil
 }
 
+// Search combines the result sets using Reciprocal Rank Fusion or Relative Score Fusion
+func HybridSubsearch(ctx context.Context, params *Params, resultSet [][]*search.Result, weights []float64, names []string, logger logrus.FieldLogger, postProc postProcFunc) ([]*search.Result, error) {
+	if len(weights) != len(resultSet) {
+		return nil, fmt.Errorf("length of weights and results do not match for hybrid search %v vs. %v", len(weights), len(resultSet))
+	}
+	if len(weights) != len(names) {
+		return nil, fmt.Errorf("length of weights and names do not match for hybrid search %v vs. %v", len(weights), len(names))
+	}
+
+	var fused []*search.Result
+	if params.FusionAlgorithm == common_filters.HybridRankedFusion {
+		fused = FusionRanked(weights, resultSet, names)
+	} else if params.FusionAlgorithm == common_filters.HybridRelativeScoreFusion {
+		fused = FusionRelativeScore(weights, resultSet, names)
+	} else {
+		return nil, fmt.Errorf("unknown ranking algorithm %v for hybrid search", params.FusionAlgorithm)
+	}
+
+	if postProc != nil {
```

Comment:
why not define the function directly here? We are defining it one level up and then pass it in and only use it here?

Reply:
Aggregate defines a different post-process function for hybrid.